test/apps/iodemo/io_demo.cc

```diff
@@ -2419,6 +2429,34 @@ class DemoClient : public P2pDemoCommon {
     MemoryPool<IoReadResponseCallback>      _read_callback_pool;
 };
 
+int set_sockaddr(const char *ip_str, uint16_t port, struct sockaddr *addr)
+{
+    struct sockaddr_in* sa_in;
+    struct sockaddr_in6* sa_in6;
+    int ret;
+
+    /* try IPv4 */
+    sa_in             = (struct sockaddr_in*)addr;
+    sa_in->sin_family = AF_INET;
+    sa_in->sin_port   = htons(port);
+    ret = inet_pton(AF_INET, ip_str, &sa_in->sin_addr);
+    if (ret == 1) {
+        return 0;
+    }
+
+    /* try IPv6 */
+    sa_in6              = (struct sockaddr_in6*)addr;
+    sa_in6->sin6_family = AF_INET6;
+    sa_in6->sin6_port   = htons(port);
+    ret = inet_pton(AF_INET6, ip_str, &sa_in6->sin6_addr);
+    if (ret == 1) {
+        return 0;
+    }
+
+    std::cout << "invalid address '" << ip_str << "'" << std::endl;
+    return -1;
```

Comment:
return bool: true in case of success, false in case of failure. same as in many ucx_wrapper.cc functions

Reply:
changed accordingly